### PR TITLE
Searchable selection for enums

### PR DESCRIPTION
**Pull request by Cooldude2606**
TL;DR If a config value is of type `"string"` and contains the `enum` property, then it will be displayed as a searchable selection dropdown. Other selection dropdowns have been made searchable.

- It was decided that only string enums would be supported until the need arises for further support.
- The following selections are now searchable: Host assignment, Default mod pack, Default user role, Mod pack version.
- Target instance during save transfer was previously searchable but used a function, this was changed to use `optionFilterProp`.
- User role assignment also used a filter function and was updated to use `optionFilterProp`.

## Changelog
```
### Changes
- String config values with whitelisted values (enum) now display as a selection dropdown. [#701](https://github.com/clusterio/clusterio/pull/701)
- Existing selection dropdowns now support search filtering. [#701](https://github.com/clusterio/clusterio/pull/701)
```

### CODE REVIEW

File: packages/web_ui/src/components/AssignInstanceModal.tsx [hunk 1]
```diff
@@ -77,13 +77,14 @@ export default function AssignInstanceModal(props: AssignInstanceModalProps) {
 			</Paragraph>
 			<Form form={form} initialValues={{ host: props.hostId ?? "null" }}>
 				<Form.Item name="host" label="Host">
-					<Select>
+					<Select showSearch optionFilterProp="name">
 						<Select.Option value={"null"}>
```
Comment (Danielv123) [inline packages/web_ui/src/components/AssignInstanceModal.tsx:81]: Should be mentioned that the <Select.Option> syntax is deprecated in antd - the new API is just an array to <Select /> instead. Not a big deal to update whenever they release v6 though.

**Review by Danielv123 — APPROVED** [review]

File: packages/web_ui/src/components/AssignInstanceModal.tsx (hunk 1 above)
Comment (Cooldude2606) [reply packages/web_ui/src/components/AssignInstanceModal.tsx:81]: Noted, I also expect other things will break when updating to v6 so we can do all changes in a single PR.
